bus-mapping/src/circuit_input_builder/transaction.rs

```diff
@@ -466,6 +466,28 @@ impl Transaction {
     }
 }
 
+#[cfg(feature = "test")]
+impl Transaction {
+    /// test if the transaction should skip post state check
+    pub fn should_skip_post_check(&self) -> bool {
+        use crate::{circuit_input_builder::execution::ExecState, precompile::PrecompileCalls};
+        let unsupported = self.steps.iter().any(|step| {
+            matches!(
+                step.exec_state,
+                ExecState::Op(OpcodeId::SELFDESTRUCT)
+                    | ExecState::Op(OpcodeId::INVALID(0xff))
+                    | ExecState::Op(OpcodeId::DIFFICULTY)
+                    | ExecState::Precompile(PrecompileCalls::Sha256)
```

Comment:
match exec_step.error = Some(ExecError::PrecompileFailed) to check hash precompiles